src/utilities/migrate-state.ts

```diff
@@ -0,0 +1,53 @@
+import { UnmigratedSerializedState, SerializedState } from "../stores/stores";
+
+export const CURRENT_SERIALIZATION_VERSION = 2;
+
+type PartialMigration = (oldState: UnmigratedSerializedState) => UnmigratedSerializedState;
+type Migration = (oldState: UnmigratedSerializedState) => SerializedState;
+
+const migrate01to02: Migration = (oldState) => {
+  if (oldState.version === 1 && !oldState.state.unit) {
+    // serailization with version = 1 and a unit specified is identical to version 2
+    const {simulation, uiStore} = oldState.state;
+    return {
+      version: 2,
+      state: {
+        unit: { name: "Tephra" },
+        blocklyStore: {
+          toolbox: simulation.toolbox,
+          initialCodeTitle: simulation. initialCodeTitle,
+          initialXmlCode: simulation.initialXmlCode
+        },
+        tephraSimulation: {
+          requireEruption: simulation.requireEruption,
+          requirePainting: simulation.requirePainting,
+          scenario: simulation.scenario,
+          stagingWindSpeed: simulation.stagingWindSpeed,
+          stagingWindDirection: simulation.stagingWindDirection,
+          stagingMass: simulation.stagingMass,
+          stagingColHeight: simulation.stagingColHeight
+        },
```

Comment:
I'm not that familiar with the details of the serialization structure, but it seems like it should be possible to do something like the following to reduce the boilerplate copying of every property:
```typescript
    const {simulation, uiStore} = oldState.state;
    const {toolbox, initialCodeTitle, initialXmlCode, ...tephraSimulation} = simulation;
    return {
      ...
      blocklyStore: {toolbox, initialCodeTitle, initialXmlCode},
	  tephraSimulation,
      ...
```
Just a thought.

Reply:
That's true. It's a little less explicit if you want to see what the final serialization (or "serailization") looks like, but it does make it clearer that the main difference between the structures is the removal of those three props from the old `tephraSimulation`, so that's nicer.